src/services/CleanupService.ts

```diff
@@ -0,0 +1,68 @@
+import {LambdaService} from "./LambdaService";
+import {ActivityService} from "./ActivityService";
+import Lambda = require("aws-sdk/clients/lambda");
+import {TestResultsService} from "./TestResultsService";
+import {
+  getActionableStaffIdsByTime,
+  getLastEventTimeByTesterStaffId,
+  getMostRecentActivityByTesterStaffId,
+  getMostRecentTestResultByTesterStaffId, getOpenVisitsToClose,
+  getStaleOpenVisits, getTesterDetailsFromTestResults,
+  getTesterStaffIds,
+  removeFromMap
+} from "../utils/helpers";
+import {IActivity, ITesterDetails, ITestResult} from "../models";
+import {TIMES} from "../utils/Enums";
+import {NotificationService} from "./NotificationService";
+
+
+export class CleanupService {
+  private lambdaService: LambdaService;
+  private activityService: ActivityService;
+  private testResultsService: TestResultsService;
+  private notificationService: NotificationService;
+  constructor(notifyService: NotificationService) {
+    this.lambdaService = new LambdaService(new Lambda());
+    this.activityService = new ActivityService(this.lambdaService);
+    this.testResultsService = new TestResultsService(this.lambdaService);
+    this.notificationService = notifyService;
+  }
+  /**
+   *
+   */
+  public async cleanupVisits(): Promise<any> {
+    // Get all activities from last period of interest
+    const allActivities: IActivity[] = await this.activityService.getRecentActivities();
+    // Get stale open visits
+    const openVisits: IActivity[] = getStaleOpenVisits(allActivities);
+    // Get list of staffIDs from open visits
+    const openVisitStaffIds: string[] = getTesterStaffIds(openVisits);
```

Comment:
Why do you need to get the unique IDs? I don't think you'll ever have multiple open visits with the same testerStaffId

Reply:
Staff IDs are required to fetch their Test Results, further down. Also used to group data coming back